subscriptions_service.go

```diff
@@ -12,13 +12,25 @@ type SubscriptionsService service
 // Update an existing subscription to specific parameter
 //
 // https://docs.lemonsqueezy.com/api/subscriptions#update-a-subscription
-func (service *SubscriptionsService) Update(ctx context.Context, params *SubscriptionUpdateParams) (*Resource[Subscription], *Response, error) {
-	response, err := service.client.do(ctx, http.MethodPatch, "/v1/subscriptions/"+params.ID, params)
+func (service *SubscriptionsService) Update(ctx context.Context, params *SubscriptionUpdateParams) (*ApiResponseSubscription, *Response, error) {
+	typeParam := "subscriptions"
+	if len(params.Type) > 0 {
```

Comment:
I think this shouldn't be optional or part of the `*SubscriptionUpdateParams` variable. You can hard code this in the payload since it won't change.